analytics/config/config.go

```diff
@@ -7,48 +7,53 @@ import (
 	"github.com/prebid/prebid-server/config"
 )
 
+type analyticsModule analytics.PBSAnalyticsModule
+
+type pbsAnalyticsModule struct {
```

Comment:
I am okay with this idea of wrapping the array of modules in a struct but what do you think about naming this struct to be something like `anlyticsModules` and the field to be called `enabled` so that when you access it, it'd be like `analyticsModules.enabled`? 
That would help communicate what it means more clearly IMHO. The name `pbsAnalyticsModule` right now is the same as the interface name `PbsAnalyticsModule` except for the upper case `P` in the front and I found that to be confusing while reading the code.

Reply:
I agree with adding clarity.

In both the current and proposed approach, we're building an array/slice of analytic modules. In the current approach, we're declaring a type alias of `[]analytics.PBSAnalyticsModule`. In the proposed approach, we're creating a new struct with a member of `[]analytics.PBSAnalyticsModule`. This is a minor difference, IMHO.

The current approach is idiomatic Go and common throughout the core libraries and in PBS. For example, BidderName is an alias of string. Perhaps renaming and reorganizing a bit can add of clarity?

Would it be clearer to rename `enabledAnalytics` to something like `Analytics` or `AnalyticsMultiplexor` and move it to the `analytics/core.go` file to sit side-by-side with the `PBSAnalyticsModule`? I also vote for renaming the `PBSAnalyticsModule` to simply `AnalyticsModule`.

Reply:
Why I've submitted this PR?

First thing, this was hard for me because I didn't practice go for a while. So, I'm a noob here :)

Here are the main "blockers"
- Not able to test easily the number of modules loaded (a cast is required)
- Not able to test the type of loaded modules, adding a type alias for each new module (FileLogger, PubstackModule ...) could be an idea
- Misleading between the interface definition and the current struct. But here, it's more related to how go is working. 
- Hard to test (fatal) when a module can't be loaded
- Hard to test if a module stops working after a while.

For the current struct or the naming, I don't have a strong opinion here.

Reply:
Ah, ok. So the essence of what you're asking for is an easier / more obvious way to inspect the slice of enabled analytics. Since the method is returning an interface, we'll always need to resort to casting in the tests. Perhaps another option is to keep the struct and break apart the interface from the list of adapter for easier to understand flow, like:

```
func NewPBSAnalytics(analytics *config.Analytics) analytics.PBSAnalyticsModule {
  return initAnalytics(analytics)
}

func initAnalytics(analytics *config.Analytics) []analytics.PBSAnalyticsModule {
  .. do all the things / test against this method ...
}
```

.. in which case I don't see much of a difference between the struct or the slice alias.

Reply:
After thinking about it again, I'm not sure that it brings much more clarity
What do you think about closing this issue?

Reply:
Sure. I'll see what I can do in the future to make it a bit cleaner. I'm fine if you close this.